File: Desafio2/desafio2.java
import java.util.Scanner;

public class desafio2 {

    /**
     * Verifica se uma palavra ou frase é um palíndromo.
     * String -> boolean
     */
    public static boolean eh_palindromo(String texto) {
        if (texto != null) {
            String tratado = texto.replaceAll("\\s+", "").toLowerCase();
            String invertido = new StringBuilder(tratado).reverse().toString();
            return tratado.equals(invertido);
        }
        return false;
    }

    public static void main(String[] args) {
        System.out.println("Bem-vindo ao verificador de palíndromos!");

        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.print("Digite uma palavra ou frase: ");
            String frase = scanner.nextLine();
            if (frase.equalsIgnoreCase("sair")) {
                break;
            }
            if (eh_palindromo(frase)) {
                System.out.println("'" + frase + "' é um palíndromo.");
            } else {
                System.out.println("'" + frase + "' não é um palíndromo.");
            }
        }

        System.out.println("Obrigado por usar o verificador de palíndromos!");
        scanner.close();
    }
}

File: Desafio2/desafio2.py
def eh_palindromo(texto):
    '''Verifica se uma palavra ou frase é um palíndromo.
    str -> bool'''
    try:
        texto_tratado = [caractere.lower() for caractere in texto if caractere.isalnum()]
        return texto_tratado == texto_tratado[::-1]
    except Exception as e:
        print(f"Erro ao processar o texto: {e}")
        return False


def main():
    '''Gerencia a interação do usuário para verificar palíndromos.
    None -> None'''

    print("Bem-vindo ao verificador de palíndromos!")

    while True:
        frase = input("Digite uma palavra ou frase: ")
        if frase.lower() == "sair":
            break
        if eh_palindromo(frase):
            print(f"'{frase}' é um palíndromo.")
        else:
            print(f"'{frase}' não é um palíndromo.")

    print("Obrigado por usar o verificador de palíndromos!")

if __name__ == "__main__":
    main()